src/seraphis_impl/serialization_demo_utils.cpp

```diff
@@ -419,6 +422,227 @@ void make_serializable_sp_destination_v1(const jamtis::JamtisDestinationV1 &dest
         sizeof(dest.addr_tag));
 }
 //-------------------------------------------------------------------------------------------------------------------
+void make_serializable_legacy_enote_variant(const LegacyEnoteVariant &enote_variant,
+    ser_LegacyEnoteVariant &serializable_enote_variant_out)
+{
+    if (enote_variant.is_type<LegacyEnoteV1>())
+    {
+        ser_LegacyEnoteV1 ser_enote;
+        LegacyEnoteV1 enote            = enote_variant.unwrap<LegacyEnoteV1>();
+        ser_enote.amount               = enote.amount;
+        ser_enote.onetime_address      = enote.onetime_address;
+        serializable_enote_variant_out = ser_enote;
+    }
+    else if (enote_variant.is_type<LegacyEnoteV2>())
+    {
+        ser_LegacyEnoteV2 ser_enote;
+        LegacyEnoteV2 enote                      = enote_variant.unwrap<LegacyEnoteV2>();
+        ser_enote.amount_commitment              = enote.amount_commitment;
+        ser_enote.onetime_address                = enote.onetime_address;
+        ser_enote.encoded_amount                 = enote.encoded_amount;
+        ser_enote.encoded_amount_blinding_factor = enote.encoded_amount_blinding_factor;
+        serializable_enote_variant_out           = ser_enote;
+    }
+    else if (enote_variant.is_type<LegacyEnoteV3>())
+    {
+        ser_LegacyEnoteV3 ser_enote;
+        LegacyEnoteV3 enote         = enote_variant.unwrap<LegacyEnoteV3>();
+        ser_enote.amount_commitment = enote.amount_commitment;
+        ser_enote.onetime_address   = enote.onetime_address;
+        memcpy(ser_enote.encoded_amount.bytes, enote.encoded_amount.bytes, sizeof(enote.encoded_amount));
+        serializable_enote_variant_out = ser_enote;
+    }
+    else if (enote_variant.is_type<LegacyEnoteV4>())
+    {
+        ser_LegacyEnoteV4 ser_enote;
+        LegacyEnoteV4 enote            = enote_variant.unwrap<LegacyEnoteV4>();
+        ser_enote.onetime_address      = enote.onetime_address;
+        ser_enote.amount               = enote.amount;
+        ser_enote.view_tag             = enote.view_tag.data;
+        serializable_enote_variant_out = ser_enote;
+    }
+    else if (enote_variant.is_type<LegacyEnoteV5>())
+    {
+        ser_LegacyEnoteV5 ser_enote;
+        LegacyEnoteV5 enote         = enote_variant.unwrap<LegacyEnoteV5>();
+        ser_enote.amount_commitment = enote.amount_commitment;
+        ser_enote.onetime_address   = enote.onetime_address;
+        memcpy(ser_enote.encoded_amount.bytes, enote.encoded_amount.bytes, sizeof(enote.encoded_amount));
+        ser_enote.view_tag             = enote.view_tag.data;
+        serializable_enote_variant_out = ser_enote;
+    }
+}
```

Comment:
In all my own programming I use to add asserts at the end of such constructs, here with a final `else`, so if we ever have `LegacyEnoteV6` and I forget to adjust this code to handle it, in addition to maybe 20 other places that have to deal with that 6th type, I will immediately notice.